src/ConstructiveSolidGeometry/SurfacePrimitives/ConeMantle.jl

```diff
@@ -87,3 +90,32 @@ function sample(c::ConeMantle{T}, Nsamps::NTuple{3,Int}) where {T}
         for φ in (Nsamps[2] ≤ 1 ? φMin : range(φMin, φMax, length = Nsamps[2]))
     ]
 end
+
+function sample(c::ConeMantle{T}, g::CylindricalTicksTuple{T})::Vector{CylindricalPoint{T}} where {T}
+    samples = [
+        CylindricalPoint{T}(get_r_at_z(c, z),φ,z)
+        for z in get_z_ticks(c, g)
+        for φ in get_φ_ticks(c, g)
+    ]
+end
+
+function _get_x_at_z(c::ConeMantle{T}, g::CartesianTicksTuple, z::T) where {T}
```

Comment:
No it is not, but `x_from_y` also results in allocations because it is stored as `Vector{T}`. Maybe we can find an allocation-free implementation for this whole thing.